context.go

```diff
@@ -37,11 +37,16 @@ func NewContext(app *App, set *flag.FlagSet, parentCtx *Context) *Context {
 
 	if c.Context == nil {
 		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			// wait for context cancellation
+			<-ctx.Done()
+			app.InterruptHandlerFunc(c)
+		}()
 		go func() {
 			defer cancel()
-			sigs := make(chan os.Signal, 1)
```

Comment:
The cancellation of the context should happen in the handler I think. You might want to completely ignore ctrl+c and that not to have any effect on the application.

Reply:
This probably requires a ctx.Cancel() or something like that to be exposed, or pass cancellation function as an argument to the handler.